blockchain/peer.go

```diff
@@ -0,0 +1,134 @@
+package blockchain
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	flow "github.com/tendermint/tendermint/libs/flowrate"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/types"
+)
+
+//--------
+// Peer
+var (
+	peerTimeout = 15 * time.Second // not const so we can override with tests
+
+	// Minimum recv rate to ensure we're receiving blocks from a peer fast
+	// enough. If a peer is not sending data at at least that rate, we
+	// consider them to have timedout and we disconnect.
+	//
+	// Assuming a DSL connection (not a good choice) 128 Kbps (upload) ~ 15 KB/s,
+	// sending data across atlantic ~ 7.5 KB/s.
+	minRecvRate = int64(7680)
+
+	// Monitor parameters
+	peerSampleRate = time.Second
+	peerWindowSize = 40 * peerSampleRate
+)
+
+type bpPeer struct {
+	logger log.Logger
+	id     p2p.ID
+
+	height      int64                  // the peer reported height
+	numPending  int32                  // number of requests still waiting for block responses
+	blocks      map[int64]*types.Block // blocks received or expected to be received from this peer
+	timeout     *time.Timer
```

Comment:
Can we have less generic name for this?

Reply:
peerTimer?

Reply:
The struct already has Peer in its name. No need to duplicate. `blockResponseTimer`?

Reply:
changed to blockResponseTimer